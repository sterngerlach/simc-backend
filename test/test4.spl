
{
    int a = 3;
    int b = 4;
    int c = 5;

    sprint("a++: ");
    iprint(a++);
    sprint("\n");

    sprint("a: ");
    iprint(a);
    sprint("\n");

    sprint("a-- + b++: ");
    iprint(a-- + b++);
    sprint("\n");

    (c)++;
    sprint("c: ");
    iprint(c);
    sprint("\n");

    sprint("c++ + c++ + c++: ");
    iprint(c++ + c++ + c++);
    sprint("\n");

    c = c++ - c--;
    sprint("c: ");
    iprint(c);
    sprint("\n");

    c = c++;
    sprint("c: ");
    iprint(c);
    sprint("\n");
}

