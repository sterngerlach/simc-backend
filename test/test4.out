a++: 3
a: 4
a-- + b++: 8
c: 6
c++ + c++ + c++: 21
c: -1
c: -1
